src/ReachSets/discretize.jl

```diff
@@ -1,310 +1,661 @@
+A matrix.
+"""
+function exp_Aδ(A::AbstractMatrix{Float64}, δ::Float64; exp_method="base")
+    if exp_method == "base"
+        return expmat(Matrix(A*δ))
+    elseif exp_method == "lazy"
+        return SparseMatrixExp(A*δ)
+    elseif exp_method == "pade"
+        return padm(A*δ)
+    else
+       throw(ArgumentError("the exponentiation method $exp_method is unknown"))
+    end
+end
 
-This uses a first order approximation of the ODE, and matrix norm upper bounds,
-see Le Guernic, C., & Girard, A., 2010, *Reachability analysis of linear systems
-using support functions. Nonlinear Analysis: Hybrid Systems, 4(2), 250-262.*
 """
-function discr_bloat_firstorder(cont_sys::InitialValueProblem{<:AbstractContinuousSystem},
-                                δ::Float64)
-
-    A, X0 = cont_sys.s.A, cont_sys.x0
-    Anorm = norm(Matrix(A), Inf)
-    ϕ = expmat(Matrix(A))
-    RX0 = norm(X0, Inf)
-
-    if inputdim(cont_sys) == 0
-        # linear case
-        α = (exp(δ*Anorm) - 1. - δ*Anorm) * RX0
-        Ω0 = CH(X0, ϕ * X0 + Ball2(zeros(size(ϕ, 1)), α))
-        return DiscreteSystem(ϕ, Ω0)
+    Φ₁(A, δ; [exp_method])
+
+Compute the series
+
+```math
+Φ₁(A, δ) = ∑_{i=0}^∞ \\dfrac{δ^{i+1}}{(i+1)!}A^i,
+```
+where ``A`` is a square matrix of order ``n`` and ``δ ∈ \\mathbb{R}_{≥0}``.
+
+### Input
+
+- `A`           -- coefficient matrix
+- `δ`           -- step size
+- `exp_method`  -- (optional, default: `"base"`) the method used to take the matrix
+                    exponential of the coefficient matrix, choose among:
+
+    - `"base"` -- the scaling and squaring method implemented in Julia base,
+                  see `?exp` for details
+    - `"pade"` -- use Pade approximant method to compute matrix exponentials of
+                  sparse matrices, implemented in `Expokit`
+    - `"lazy"` -- compute a wrapper type around the matrix exponential, i.e. using
+                  the lazy implementation `SparseMatrixExp` from `LazySets` and
+                  the evaluation of the action of the matrix exponential using the
+                  `expmv` implementation from `Expokit`
+### Output
+
+A matrix.
+
+### Algorithm
+
+We use the method from [1]. If ``A`` is invertible, ``Φ₁`` can be computed as
+
+```math
+Φ₁(A, δ) = A^{-1}(e^{δA} - I_n).
+```
+
+In the general case, implemented in this function, it can be computed as
+submatrices of the block matrix
+
+```math
+P = \\exp \\begin{pmatrix}
+Aδ && δI_n && 0 \\
+0 && 0 && δI_n \\
+0 && 0 && 0
+\\end{array}.
+```
+It can be shown that `Φ₁(A, δ) = P[1:n, (n+1):2*n]`.
+
+[1] Frehse, Goran, et al. "SpaceEx: Scalable verification of hybrid systems."
+International Conference on Computer Aided Verification. Springer, Berlin,
+Heidelberg, 2011.
+"""
+function Φ₁(A, δ; exp_method="base")
+    n = size(A, 1)
+    if exp_method == "base"
+        P = expmat(Matrix([A*δ     sparse(δ*I, n, n)  spzeros(n, n);
+                   spzeros(n, 2*n) sparse(δ*I, n, n);
+                   spzeros(n, 3*n)]))
```

Comment:
Can you create a helper function to define this matrix? It is identical to the ones below (even in Φ2).